litejpeg/core/rle/entropycoder.py

```diff
@@ -0,0 +1,219 @@
+# Entropycoder Module
+
+# Importing libraries.
+from litex.gen import *
+from litex.soc.interconnect.stream import *
+
+from litejpeg.core.common import *
+
+"""
+Entropycoder Module:
+----------------
+This module is been made in order to calculate the number of bits to
+store the Amplitude.
+
+Parameters:
+-----------
+input_data :
+    Contains the input data coming from the other modules or the Test Benches.
+size :
+    Contains the number of bits in the Amplitude.
+"""
+
+
+# To provide delay so in order to sink the data coming from the main
+# module to that of the Datapath module.
+datapath_latency = 3
+
+
+@CEInserter()
+class EntropyDatapath(Module):
+
+    """
+    EntropyDatapath :
+    ------------------
+    It contains the steps for the Entropycoder to calculate the bits for the
+    Amplitude to store.
+
+    Parameters:
+    -----------
+    sink : Get the input from the other modules.
+           The input is the modulus of the input.
+    source : Give the output to the other modules.
+    input_data : modulus of input of the matrix.
+    get_data : Contains the temporary data for calculating the size.
+    size : The number of bits required for storing th amplitude.
+
+    """
+
+    def __init__(self):
+
+        # Record the input and output of the Datapath.
+        # sink = input
+        # source = output
+        self.sink = sink = Record(block_layout(12))
+        self.source = source = Record(block_layout(4))
+
+        # Getting the input.
+        input_data = Signal(12)
+
+        # The values for storing the temporary values for
+        # calculating the size.
+        get_data = Array(Signal(12) for i in range(12))
+
+        # Store the output.
+        size = Signal(4)
+
+        # For calculating the size.
+
+        # Storing the size in the input_data.
```

Comment:
Your comments should concentrate on *why* you are doing something. The code explains how you are doing it.

For example, why do you want to iterate over different values from 1 to 12? Any python programmer understands that `for i in range(12)` iterates over the values 0->11.

Reply:
DONE